src/app/sh-http/sh-http.service.test.js

```diff
@@ -78,4 +94,20 @@ describe('ShHttpService', () => {
         sut.delete(urlMock);
         expect(httpMock.delete).toHaveBeenCalledWith(urlMock, options);
     });
+
+    it('should catch error', () => {
+        const urlMock = 'mock';
+        let options = new ShRequestOptions({
+            headers: new Headers({
+                'Content-Type': 'application/json'
+            })
+        });
+        options = options.merge({
+            method: RequestMethod.Delete,
+            url: urlMock
+        });
+        errorResponse = true;
+        sut.delete(urlMock);
+        expect(sut._errorHandling).toHaveBeenCalled();
```

Comment:
!fix
its a bad practice to check sut's function call: that might be internal be hidden from outer world. or next refactoring will split that function into two.